### PR TITLE
audio-hal: Calibration extravaganza

### PR DESCRIPTION
**Pull request by kholk**
Just remember to remove that draft status from the github PR to sonyxperiadev after merging this... :)))

### CODE REVIEW

File: hal/audio_extn/cirrus_sony.c [hunk 1]
```diff
@@ -615,28 +626,28 @@ static int cirrus_play_silence(int seconds) {
...
     }
 
-    while (!adev->primary_output) {
-        ALOGE("Still no primary_output!");
-        // TODO: Perhaps wait on a condvar like spkr_prot?
-        usleep(1000);
+    while ((!adev->primary_output || !adev->platform) && adev_retry) {
+        ALOGI("%s: Waiting for audio device...", __func__);
+        sleep(1);
+	adev_retry--;
```
Comment (MarijnS95) [inline hal/audio_extn/cirrus_sony.c:632]: Fix your indentation, yo!

File: hal/audio_extn/cirrus_sony.c [hunk 2]
```diff
@@ -445,7 +456,7 @@ static int cirrus_set_mixer_value_by_name(char* ctl_name, int value) {
 
     ctl_config = mixer_get_ctl_by_name(card_mixer, ctl_name);
     if (!ctl_config) {
-        ALOGE("%s: Cannot get mixer control %s", __func__, ctl_name);
+        ALOGD("%s: Cannot get mixer control %s", __func__, ctl_name);
```
Comment (MarijnS95) [inline hal/audio_extn/cirrus_sony.c:459]: We should find a better way to be clever about mono vs stereo, seeing these warnings obviously in the logs is vital in most other scenarios.

File: hal/audio_extn/cirrus_sony.c [hunk 3]
```diff
@@ -615,28 +626,28 @@ static int cirrus_play_silence(int seconds) {
...
     uc_info_rx->out_snd_device = SND_DEVICE_OUT_SPEAKER;
     list_add_tail(&adev->usecase_list, &uc_info_rx->list);
+
     fp_platform_check_and_set_codec_backend_cfg(adev, uc_info_rx,
                                              uc_info_rx->out_snd_device);
+
     fp_enable_snd_device(adev, uc_info_rx->out_snd_device);
     fp_enable_audio_route(adev, uc_info_rx);
 
+    select_devices(adev, uc_info_rx->id);
```
Comment (MarijnS95) [inline hal/audio_extn/cirrus_sony.c:649]: Whoa, that's a massively complex function! In the end it supersedes the call to `fp_enable_snd_device` + `fp_enable_audio_route`, all we need then is `fp_platform_check_and_set_codec_backend_cfg` afaik.

Anyway, as mentioned above it is not available to `audio_extn` libs.

File: hal/audio_extn/cirrus_sony.c [hunk 4]
```diff
@@ -645,25 +656,15 @@ static int cirrus_play_silence(int seconds) {
...
-        goto exit;
-    }
-
-    if (!pcm_is_ready(handle.pcm_rx)) {
-        ALOGE("%s: The PCM device is not ready: %s", __func__,
-              pcm_get_error(handle.pcm_rx));
-        ret = -EINVAL;
-        goto exit;
-    }
+    handle.pcm_rx = handle.pcm_tx = NULL;
```
Comment (MarijnS95) [inline hal/audio_extn/cirrus_sony.c:659]: Lmao

**Review by MarijnS95 — CHANGES_REQUESTED** [review]
Guess why these are not in any header, but (some!) only exposed through `spkr_prot_init_config_val.fp_*`...
```c
ld.lld: error: undefined symbol: select_devices
>>> referenced by cirrus_sony.c:649 (vendor/qcom/opensource/audio-hal/primary-hal/hal/audio_extn/cirrus_sony.c:649)
>>>               out/target/product/pdx206/obj/SHARED_LIBRARIES/libcirrusspkrprot_intermediates/cirrus_sony.o:(cirrus_play_silence)

ld.lld: error: undefined symbol: pcm_open_prepare_helper
>>> referenced by cirrus_sony.c:662 (vendor/qcom/opensource/audio-hal/primary-hal/hal/audio_extn/cirrus_sony.c:662)
>>>               out/target/product/pdx206/obj/SHARED_LIBRARIES/libcirrusspkrprot_intermediates/cirrus_sony.o:(cirrus_play_silence)

ld.lld: error: undefined symbol: platform_invalidate_backend_config
>>> referenced by cirrus_sony.c:704 (vendor/qcom/opensource/audio-hal/primary-hal/hal/audio_extn/cirrus_sony.c:704)
>>>               out/target/product/pdx206/obj/SHARED_LIBRARIES/libcirrusspkrprot_intermediates/cirrus_sony.o:(cirrus_play_silence)
```

@kholk can you please check the "Allow edits from maintainers" so that I can push a rebased version of this PR? GH is lost and makes it a mess now, because I squashed+force-pushed some cleanups into my own commits (where they belong).

**Comment by kholk** [comment]
> Guess why these are not in any header, but (some!) only exposed through `spkr_prot_init_config_val.fp_*`...
> 
> ```c
> ld.lld: error: undefined symbol: select_devices
> >>> referenced by cirrus_sony.c:649 (vendor/qcom/opensource/audio-hal/primary-hal/hal/audio_extn/cirrus_sony.c:649)
> >>>               out/target/product/pdx206/obj/SHARED_LIBRARIES/libcirrusspkrprot_intermediates/cirrus_sony.o:(cirrus_play_silence)
> 
> ld.lld: error: undefined symbol: pcm_open_prepare_helper
> >>> referenced by cirrus_sony.c:662 (vendor/qcom/opensource/audio-hal/primary-hal/hal/audio_extn/cirrus_sony.c:662)
> >>>               out/target/product/pdx206/obj/SHARED_LIBRARIES/libcirrusspkrprot_intermediates/cirrus_sony.o:(cirrus_play_silence)
> 
> ld.lld: error: undefined symbol: platform_invalidate_backend_config
> >>> referenced by cirrus_sony.c:704 (vendor/qcom/opensource/audio-hal/primary-hal/hal/audio_extn/cirrus_sony.c:704)
> >>>               out/target/product/pdx206/obj/SHARED_LIBRARIES/libcirrusspkrprot_intermediates/cirrus_sony.o:(cirrus_play_silence)
> ```
> 
> @kholk can you please check the "Allow edits from maintainers" so that I can push a rebased version of this PR? GH is lost and makes it a mess now, because I squashed+force-pushed some cleanups into my own commits (where they belong).

It's checked since year 0.... :P

**Comment by MarijnS95** [comment]
@kholk Except that it isn't? github says:

```
Add more commits by pushing to the generic-usecase branch on kholk/vendor-qcom-opensource-audio-hal-primary-hal.
```

But github also says:

```console
me:primary-hal/ $ gpf git@github.com:kholk/vendor-qcom-opensource-audio-hal-primary-hal.git HEAD:generic-usecase
ERROR: Permission to kholk/vendor-qcom-opensource-audio-hal-primary-hal.git denied to MarijnS95.
fatal: Could not read from remote repository.

Please make sure you have the correct access rights
and the repository exists.
```

**Comment by MarijnS95** [comment]
I even tried updating the PR directly:

```
me:audio-hal/ $ gpf origin HEAD:refs/pull/1/head
...
To github.com:MarijnS95/vendor-qcom-opensource-audio-hal-primary-hal.git
 ! [remote rejected]     HEAD -> refs/pull/1/head (deny updating a hidden ref)
error: failed to push some refs to 'github.com:MarijnS95/vendor-qcom-opensource-audio-hal-primary-hal.git'
```

**Pull request closed without merging**